tools/test-generator/src/main/java/org/citrusframework/generate/provider/http/HttpCodeProvider.java

```diff
@@ -43,7 +43,7 @@ void provideRequestConfiguration(final CodeBlock.Builder code, final HttpMessage
     }
 
     void provideResponseConfiguration(final CodeBlock.Builder code, final HttpMessage message) {
-        code.add(".response($T.$L)\n", HttpStatus.class, message.getStatusCode().name());
+        code.add(".response($T.$L)\n", HttpStatus.class, message.getStatusCode().value());
```

Comment:
As far as i see, name() returns a string, wheras value() returns an int.
Not exactly sure what this method does but you could do better here making an instanceof check.

Resolve to HttpStatusCode, check if it is instanceof HttpStatus and if so, get the name. Otherwise chose a default. E,g "Status<YourCustomStatusCode>"